rosidl_generator_py/resource/_msg.py.em

```diff
@@ -181,45 +181,58 @@ class @(spec.base_type.type)(metaclass=Metaclass):
+@{assert_msg_suffixes.insert(1, 'with length <= %d' % field.type.array_size)}@
 @[      else]@
              len(value) == @(field.type.array_size) and
+@{assert_msg_suffixes.insert(1, 'with length %d' % field.type.array_size)}@
 @[      end if]@
 @[    end if]@
              all([isinstance(v, @(get_python_type(field.type))) for v in value]) and
-@[      if field.type.type.startswith('int')]@
+@{assert_msg_suffixes.append("and each value of type '%s'" % get_python_type(field.type))}@
+@[    if field.type.type.startswith('int')]@
 @{
 nbits = int(field.type.type[3:])
 bound = 2**(nbits - 1)
 }@
-             all([val >= -@(bound) and val < @(bound) for val in value]))
-@[      elif field.type.type.startswith('uint')]@
+             all([val >= -@(bound) and val < @(bound) for val in value])), \
+@{assert_msg_suffixes.append('and each integer in [%d, %d)' % (-bound, bound))}@
+@[    elif field.type.type.startswith('uint')]@
 @{
 nbits = int(field.type.type[4:])
 bound = 2**nbits
 }@
-             all([val >= 0 and val < @(bound) for val in value]))
-@[      elif field.type.type == 'char']@
-             all([ord(val) >= -128 and ord(val) < 128 for val in value]))
-@[      else]@
-             True)
-@[      end if]@
+             all([val >= 0 and val < @(bound) for val in value])), \
+@{assert_msg_suffixes.append('and each unsigned integer in [0, %d)' % bound)}@
+@[    elif field.type.type == 'char']@
+             all([ord(val) >= -128 and ord(val) < 128 for val in value])), \
+@{assert_msg_suffixes.append('and each characters ord() in [-128, 128)')}@
+@[    else]@
+             True), \
+@[    end if]@
+            "The '@(field.name)' field must be @(' '.join(assert_msg_suffixes))"
 @[  elif field.type.string_upper_bound]@
             ((isinstance(value, str) or isinstance(value, UserString)) and
-             len(value) <= @(field.type.string_upper_bound))
+             len(value) <= @(field.type.string_upper_bound)), \
+            "The '@(field.name)' field must be string value not longer than @(field.type.string_upper_bound)"
 @[  elif not field.type.is_primitive_type()]@
-            isinstance(value, @(field.type.type))
+            isinstance(value, @(field.type.type)), \
+            "The '@(field.name)' field must be a sub message of type '@(field.type.type)'"
 @[  elif field.type.type == 'byte']@
             ((isinstance(value, bytes) or isinstance(value, ByteString)) and
-             len(value) == 1)
+             len(value) == 1), \
+            "The '@(field.name)' field must of type 'bytes' or 'ByteString' with a length 1"
 @[  elif field.type.type == 'char']@
             ((isinstance(value, str) or isinstance(value, UserString)) and
-             len(value) == 1 and ord(value) >= -128 and ord(value) < 128)
+             len(value) == 1 and ord(value) >= -128 and ord(value) < 128), \
+            "The '@(field.name)' field must of type 'str' or 'UserString' with a length 1 and the character ord() in [-128, 127)"
```

Comment:
I know we don't check line length on generated code but for constant strings like that I think wrapping would improved readability